### ISSUE TITLE
parallel test failed sometimes on a single core machine

### ISSUE THREAD

**Issue by 31z4** [meta-toolkit/meta#76]
To reproduce Vagrant or just VirtualBox with single core can be used.

```
ctest --output-on-failure
Test project /home/vagrant/meta/build
      Start  1: analyzers
 1/14 Test  #1: analyzers ........................   Passed    0.03 sec
      Start  2: stemmers
 2/14 Test  #2: stemmers .........................   Passed    0.53 sec
      Start  3: parallel
 3/14 Test  #3: parallel .........................***Failed    5.44 sec
        parallel-speed: [ FAIL ] [parallel_time.count() < serial_time.count()] => [2633 < 2587] (parallel_test.cpp:43)

      Start  4: inverted-index
 4/14 Test  #4: inverted-index ...................   Passed    7.33 sec
      Start  5: forward-index
 5/14 Test  #5: forward-index ....................   Passed    8.24 sec
      Start  6: string-list
 6/14 Test  #6: string-list ......................   Passed    0.01 sec
      Start  7: vocabulary-map
 7/14 Test  #7: vocabulary-map ...................   Passed    0.01 sec
      Start  8: libsvm-parser
 8/14 Test  #8: libsvm-parser ....................   Passed    0.00 sec
      Start  9: classifiers
 9/14 Test  #9: classifiers ......................   Passed   16.79 sec
      Start 10: rankers
10/14 Test #10: rankers ..........................   Passed   32.04 sec
      Start 11: ir-eval
11/14 Test #11: ir-eval ..........................   Passed    3.18 sec
      Start 12: compression
12/14 Test #12: compression ......................   Passed    0.05 sec
      Start 13: graph
13/14 Test #13: graph ............................   Passed    0.01 sec
      Start 14: parser
14/14 Test #14: parser ...........................   Passed    0.01 sec

93% tests passed, 1 tests failed out of 14

Total Test time (real) =  73.73 sec

The following tests FAILED:
          3 - parallel (Failed)
Errors while running CTest
```

**Comment by skystrife** [comment]
Right. We expect the parallelized version to be faster than the serial version, so it makes sense that this would break on single core systems/VMs. We could probably conditionally enable this test only if `std::thread::hardware_concurrency() > 1`... What do you think, @smassung?

**Comment by smassung** [comment]
Yes, I agree, and just committed a fix in eee7dfe76bc108703ae630339c797f11259eb696. I only conditionally enable the speed parallel test, and leave the others since those tests would still be useful. Thanks, @31z4
